### PR TITLE
Game Development Resources Added

### LINKED ISSUE #888: 💡 [FEATURE] - Add Game Development Resources
### Idea Contribution

- [X] I have read all the feature request issues.
- [X] I'm interested in working on this issue
- [X] I'm part of GSSOC organization

### Explain feature request

Game developers need a centralized platform offering:

- Tutorials: Step-by-step guides for various skill levels.

- Asset Libraries: Pre-made graphics, sounds, and models.

- Development Tools: Game engines, IDEs, and plugins.

- Frameworks and Libraries: Simplifying code resources.

- Documentation: Detailed, updated guides.

- Community Support: Forums, chat groups, and Q&A sites.

- Project Templates: Ready-to-use templates.

- Optimization Guides: Performance improvement resources.

### Explain your solution

Create a centralized online platform serving as a comprehensive hub for game development resources. This platform will include interactive tutorials with categorized guides and video content, a searchable asset library with pre-made assets, and a tool repository with user reviews and download links. It will offer a collection of frameworks and libraries with example projects, a documentation archive, and a community forum with discussion boards and chat groups.

### Any alternative approaches/features

1.Ensure that all links are active and lead to relevant, high-quality resources.
2.Aim to include a mix of beginner, intermediate, and advanced resources to cater to a wide range of learners.

### Additional Context

I want to contribute in this issue under GSSoC'24 , could you assign this issue to me please @jfmartinz

### PR DESCRIPTION
**Pull request by shuvadeepmondal**
# Pull Request
- ` Game Development Resources `

## Description
 I have added the "Game Development" resources . There are some contents about the Roadmaps, Tools, Techniques , Books, 
 YouTube Channels & GitHub repository at the Resource page.

Closes #888  

## Type of change
- [x] Adding a new Resource
- [x] Adding a new Category
- [ ] Documentation enhancement or fixes
- [ ] Website enhancement or fixes

## Checklist:
- [x] I read carefully [CONTRIBUTING.md](https://github.com/jfmartinz/ResourceHub/blob/main/CONTRIBUTING.md)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: Game Development/readme.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+- [YouTube Channels](#youtube-channels)
+- [GitHub Repositories](#github-repositories)
+
+### Roadmap
+
+> To become a proficient Game Developer, there are various skills and areas of knowledge you need to master. Here’s a roadmap to guide you through the learning process:
+
+| Resource Name | Description |
+| ------------- | ----------- |
+| [Game Development Roadmap](https://roadmap.sh/game) | Start your game development journey with roadmap.sh's guide. This comprehensive resource covers basic to advanced topics, including game design, programming, art creation, and testing. Featuring best practices, tutorials, interactive videos, and community support, it's ideal for all skill levels. |
```
Comment (jfmartinz) [inline Game Development/readme.md:20]: Follow the format

File: Game Development/readme.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+| Resource Name | Description | Cost |
+| ------------- | ----------- | ---- |
+| [Udemy Game Development Courses](https://www.udemy.com/topic/game-development/) | Offers a variety of courses on game development, design, and programming. | Paid |
+| [Coursera Game Development Courses](https://www.coursera.org/courses?query=game%20development) | Provides online courses on game development, often created by universities and colleges. | Paid |
+| [edX Game Development Courses](https://www.edx.org/learn/game-development) | Hosts a range of game development courses from top institutions, covering various aspects of game creation. | Paid |
+| [Codecademy - Learn Game Development](https://www.codecademy.com/learn/paths/create-video-games-with-phaser.js) | Interactive course that teaches game development basics using the Phaser.js framework. | Paid |
+| [GameDev.tv](https://www.gamedev.tv/) | Platform offering a wide range of game development courses, including Unity, Unreal Engine, and more. | Paid |
+| [freeCodeCamp - Game Development](https://www.freecodecamp.org/news/tag/game-development/) | Provides free tutorials and articles on game development concepts and techniques. | Free |
```
Comment (jfmartinz) [inline Game Development/readme.md:135]: Follow the format

File: Game Development/readme.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+### GitHub Repositories
+
+> You can follow these GitHub Repositories for Game Development:
+
+| Resource Name | Description |
+| ------------- | ----------- |
+| [Awesome Game Development](https://github.com/mbrukman/awesome-game-development) | Curated list of game development resources, tools, and libraries. |
+| [Game Development Resources](https://github.com/collections/game-development) | GitHub's collection of game development resources and repositories. |
+| [Unity Tech Demos](https://github.com/Unity-Technologies/UnityCsReference) | Unity Technologies' repository of tech demos and sample projects. |
```
Comment (jfmartinz) [inline Game Development/readme.md:167]: Follow the format

**Review by jfmartinz — CHANGES_REQUESTED** [review]

File: Game Development/readme.md (hunk 3 above)
Comment (shuvadeepmondal) [reply Game Development/readme.md:167]: Okay Sir

**Comment by shuvadeepmondal** [comment]
@jfmartinz Sir i have done the changes

**Review by jfmartinz — APPROVED** [review]

**Pull request merged**